build.gradle

```diff
@@ -71,144 +72,174 @@ configurations {
-    archiveZip.getInputStream(archiveZip.getEntry('META-INF/MANIFEST.MF')).withStream {
-      assert it.text.contains("Plugin-Version: ${project.version}"): "Wrong metadate in file ${archive} - run a clean build"
+    new ZipFile(archive).withCloseable { archiveZip ->
+        archiveZip.getInputStream(archiveZip.getEntry('META-INF/MANIFEST.MF')).withStream {
+            assert it.text.contains("Plugin-Version: ${project.version}"): "Wrong metadate in file ${archive} - run a clean build"
+        }
     }
-  }
 }
 
 tasks.withType(AbstractArchiveTask) {
-  inputs.property('pluginVersion') {
-    project.version
-  }
+    inputs.property('pluginVersion') {
+        project.version
+    }
 }
 
 task checkArchiveManifests {
-  dependsOn jar, jpi
-  doLast {
-    checkArchiveManifest(jar.archiveFile.get().asFile)
-    checkArchiveManifest(jpi.archiveFile.get().asFile)
-  }
+    dependsOn jar, jpi
+    doLast {
+        checkArchiveManifest(jar.archiveFile.get().asFile)
+        checkArchiveManifest(jpi.archiveFile.get().asFile)
+    }
 }
 
 tasks.withType(AbstractPublishToMaven) {
-  dependsOn checkArchiveManifests
+    dependsOn checkArchiveManifests
 }
 
 defaultTasks.add 'test'
 defaultTasks.add 'jpi'
 
 task createWrapperZip(type: Zip) {
-  archiveFileName = 'wrapper.zip'
-  destinationDirectory = new File(sourceSets.test.output.resourcesDir, 'gradle')
+    archiveFileName = 'wrapper.zip'
+    destinationDirectory = new File(sourceSets.test.output.resourcesDir, 'gradle')
 
-  from(project.rootDir) {
-    include('gradle/**')
-    include('gradlew*')
-  }
+    from(project.rootDir) {
+        include('gradle/**')
+        include('gradlew*')
+    }
 }
 
 processTestResources.dependsOn(createWrapperZip)
 
 task generateVersionFile() {
-  var versionsDir = layout.buildDirectory.dir('versions')
-  doLast {
-      def outputDir = versionsDir.get().asFile
-      outputDir.mkdirs()
+    var versionsDir = layout.buildDirectory.dir('versions')
+    doLast {
+        def outputDir = versionsDir.get().asFile
+        outputDir.mkdirs()
 
-      new File(outputDir, "gradle-enterprise-maven-extension-version.txt").text = "${gradleEnterpriseMavenExtensionVersion}"
-      new File(outputDir, "common-custom-user-data-maven-extension-version.txt").text = "${commonCustomUserDataMavenExtensionVersion}"
-  }
+        new File(outputDir, "gradle-enterprise-maven-extension-version.txt").text = "${gradleEnterpriseMavenExtensionVersion}"
+        new File(outputDir, "common-custom-user-data-maven-extension-version.txt").text = "${commonCustomUserDataMavenExtensionVersion}"
+    }
+
+    inputs.property('gradleEnterpriseMavenExtensionVersion', gradleEnterpriseMavenExtensionVersion)
+    inputs.property('commonCustomUserDataMavenExtensionVersion', commonCustomUserDataMavenExtensionVersion)
+
+    outputs.dir(versionsDir)
+        .withPropertyName('versionsDir')
+}
+
+def mavenVersions = [
+// Uncomment after 1.15.3 is released
+//    '3.1.1',
+//    '3.2.1',
```

Comment:
Is there anything specific in `3.2.1`? We can just use the latest patch release from `3.2.5`.

Reply:
Make sense.